standards/core/mix.md

```diff
@@ -0,0 +1,119 @@
+
+- An `entry` node is the originator node of a message, i.e a node that wishes to publish/query messages to/from the waku network.
+- An `exit` node is responsible for delivery messages to destination peer in the network.
+- An `intermediary` node is responsible for just forwarding mix packet to the next mix node in the path.
+
+All the relay nodes would by default act as mix `intermediary` and `exit` nodes in the network. Any node that wishes to publish/query messages from the waku network would act as an entry node. The implementation can provide a configuration to disable a node from acting as an `intermediary` node.
+
+Light/Edge nodes will only be acting as `entry` nodes.
+
+## Flow
+
+TBD
+
+## ENR updates
+
+Each waku node that supports mix shall include the same in its ENR as per [Waku ENR](https://github.com/waku-org/specs/blob/master/standards/core/enr.md). 
+The `bit 5` in the [waku2 ENR key](https://github.com/waku-org/specs/blob/master/standards/core/enr.md#waku2-enr-key) shall be used to indicate a node's support for mix protocol.
+
+Each mix node shall also include its `ed25519 public key` that is used for sphinx encryption as a separate field `mix-key` in the [Waku ENR](https://github.com/waku-org/specs/blob/master/standards/core/enr.md).
+
+
+### Discovery
+
+Mix protocol provides better anonymity when an originator node has a large pool of mix nodes to choose randomly from. This moves the problem into discover domain and requires the following from discovery mechanisms:
+1. it is important for nodes to be able to discover as many nodes as possible quickly. This becomes especially important for edge nodes that come online just to publish/query messages for a short period of time.
+2. it is important to have the most recent online status of the nodes so that broken mix paths are not selected leading to reliability issues.
+
+Point-2 above can be mitigated partially by choosing redundant mix paths for the same message by the originator but this may not be effective.
+
+Combining existing discovery mechanisms of [WAKU DISCV5](https://rfc.vac.dev/waku/standards/core/33/discv5), [WAKU PEER-EXCHANGE](https://rfc.vac.dev/waku/standards/core/34/peer-exchange) and [WAKU RENDEZVOUS](https://github.com/waku-org/specs/blob/master/standards/core/rendezvous.md) should help address this problem till a certain extent, but not completely. 
+
+As Waku rendezvous proposes to use a 10 second registration interval and a 1 minute expiry for all nodes to a rendezvous point, this should help nodes maintain up-to-date online status of other mix nodes in the network.
+
+Edge nodes will have to use rendezvous discovery in order to discover a larger and available set of mix nodes quickly.
+
+Having following improvements in discovery would further improve the efficacy of mix protocol:
+- quicker discovery of most of the mix nodes
+- to be aware of online status of the mix nodes.
```

Comment:
I think it's OK to have a "Discovery" section in the raw spec where we state the problem (as you've done above), but I would remove any of the existing implementation details (i.e. line 72 onwards). It's OK to leave unknown sections unspecified, rather than reporting on existing shortcomings (which would date the spec quickly)